src/vm.sh

```diff
@@ -59,12 +94,71 @@ function vm_ssh
   ssh -p ${configurations[port]} ${configurations[ip]}
 }
 
+function vm_clean
+{
+  say "Cleaning kw-related libvirt entries..."
+  vm_down
+
+  
+  if [ $(net_status $VIRT_NET_NAME) == "inactive" ]; then
+    say "Undefining kw libvirt network with: $VIRT_NET_UNDEFINE $VIRT_NET_NAME"
+    $VIRT_NET_UNDEFINE $VIRT_NET_NAME
+  fi
+  if [ $(vm_status $VIRT_VM_NAME) == "shut" ]; then
+    say "Undefining kw vm with: $VIRT_UNDEFINE $VIRT_VM_NAME"
+    $VIRT_UNDEFINE $VIRT_VM_NAME
+  fi
+}
+
+function vm_register
+{
+
+  check_local_configuration
+  
+  local current_path=$PWD
+  local virt_net_config_path=$PWD/kworkflow-network.xml
+
+  #set network config to default if inexistent
+  if [ ! -f $virt_net_config_path ]; then
+    virt_net_config_path=$DEFAULT_CONFIG_PATH/kworkflow-network-default.xml
+  fi
+
+  if [ $(vm_status $VIRT_VM_NAME) != "inexistent" ] || \
```

Comment:
1. Remove the VM name validation from here (see the next review comment)
2. Check if we already have the network and, based on that, take some decision. We suggest something like this:

```sh
if [ $(net_status $VIRT_NET_NAME) != "inexistent" ]  ; then
  say "VM installation process."
  say "Creating libvirt network. Sudo is required here."
  sudo $VIRT_NET_DEFINE --file $virt_net_config_path  
  $VIRT_NET_START $VIRT_NET_NAME
fi
```